composer.json

```
@@ -63,9 +63,22 @@
     "minimum-stability": "dev",
     "prefer-stable": true,
     "scripts": {
-        "auto-scripts": {
-            "cache:clear": "symfony-cmd",
-            "assets:install %PUBLIC_DIR%": "symfony-cmd"
-        }
+        "build:docs": [
+            "composer bin docs install",
+            "bin/tools/docs/vendor/symfony-tools/docs-builder/bin/console build:docs --fail-on-errors docs/ docs/output/"
+        ],
+        "build:fixcs": "echo TODO",
+        "build:sca": "echo TODO",
+        "test:full": [
+            "DATABASE_URL=mysql://root:root@127.0.0.1:33061/foundry MONGO_URL=mongodb://127.0.0.1:27017/dbName?compressors=disabled&amp;gssapiServiceName=mongodb USE_FOUNDRY_BUNDLE=1 vendor/bin/simple-phpunit --configuration phpunit-dama-doctrine.xml.dist"
+        ],
```

I'd like to be able to have this:

```suggestion
        "test:full": [
            "docker composer up -d",
            "(wait for mysql/mongo to be available??)"
            "DATABASE_URL=mysql://root:root@127.0.0.1:33061/foundry MONGO_URL=mongodb://127.0.0.1:27017/dbName?compressors=disabled&amp;gssapiServiceName=mongodb USE_FOUNDRY_BUNDLE=1 vendor/bin/simple-phpunit --configuration phpunit-dama-doctrine.xml.dist"
            "docker composer down",
        ],
```

---

> "(wait for mysql/mongo to be available??)"

This can be achieved using [healthchecks](https://docs.docker.com/compose/compose-file/compose-file-v3/#healthcheck) in the Docker compose.

---

Right, but I wasn't sure how to actually have docker `compose up -d` _pause_ until the health check passes. I was wondering if I could create a dummy service that `depends_on` `mysql` and `mongo`. Something like:

```yaml
services:
  dummy:
    depends_on:
      mysql:
        condition: service_healthy
      mongo:
        condition: service_healthy
```

---

I'm actually wondering if we should not provide a php container. It has several benefits:
- It could be used to run the healthcheck
- it would be easier to contribute because all needed extensions will be already installed (mongo, pgsql, mysql...)
- we could fix the version for sttic analysis

WDYT?

---

I think this makes sense. It also would prevent users from using code that's incompatible with php 7.2.

@wouterj, wdyt?